source/compiler-core/slang-command-line-args.h

```diff
@@ -0,0 +1,95 @@
+#ifndef SLANG_COMMAND_LINE_ARGS_H
+#define SLANG_COMMAND_LINE_ARGS_H
+
+// This file defines the `Name` type, used to represent
+// the name of types, variables, etc. in the AST.
+
+#include "../core/slang-basic.h"
+
+#include "slang-source-loc.h"
+#include "slang-diagnostic-sink.h"
+
+namespace Slang {
+
+struct CommandLineArg
+{
+    String value;           ///< The value of the arg
+    SourceLoc loc;          ///< The location of the arg
+};
+
+struct CommandLineArgs
+{
+    typedef CommandLineArg Arg;
+
+    SLANG_FORCE_INLINE Index getArgCount() const { return m_args.getCount(); }
+    const Arg& operator[](Index i) const { return m_args[i]; }
+
+    const Arg* begin() const { return m_args.begin(); }
+    const Arg* end() const { return m_args.end(); }
+
+        /// NOTE! Should NOT include the executable name
+    void setArgs(const char*const* args, size_t argCount);
+
+        /// Ctor with a source manager
+    CommandLineArgs(SourceManager* manager):
+        m_sourceManager(manager),
+        m_sourceView(nullptr)
+    {
+    }
+
+    String m_executablePath;                ///< Can be optionally be set
+
+    List<Arg> m_args;                       ///< The args
+    SourceManager* m_sourceManager;         ///< The source manager and associated diagnostics sink
+    SourceView* m_sourceView;        ///< contains the command line as source
+};
+
+struct CommandLineReader
+{
+        /// Peek the current location
+    SourceLoc peekLoc() const { return m_index < m_args->getArgCount() ? (*m_args)[m_index].loc : SourceLoc(); }
```

Comment:
Ideally we'd want to be able to indicate a location "one past the end" similar to how the EOF token lets us have error messages that point to the end of the argument list.

Assuming that the synthesized command-line string is registered as a source file, we should be able to produce a location that represents its last possible location.